src/Symfony/Component/Messenger/Asynchronous/Routing/SenderLocator.php

```diff
@@ -32,13 +32,25 @@ public function __construct(ContainerInterface $senderServiceLocator, array $mes
      */
     public function getSendersForMessage($message): array
     {
-        $senderIds = $this->messageToSenderIdsMapping[\get_class($message)] ?? $this->messageToSenderIdsMapping['*'] ?? array();
-
         $senders = array();
-        foreach ($senderIds as $senderId) {
+        foreach ($this->getSenderIds($message) as $senderId) {
             $senders[] = $this->senderServiceLocator->get($senderId);
         }
 
         return $senders;
     }
+
+    private function getSenderIds($message): array
+    {
+        $senderIds = array();
```

Comment:
I'm not sure you need that variable anymore, you can use `return`s directly.

Reply:
Indeed.